src/controllers/token.ts

```diff
@@ -31,6 +31,33 @@ export default class TokenController {
     }
   }
 
+  @Post('/waitlist')
+  async addWaitlistToken(
+    @Body({ required: true }) body: TokenWithSettings,
+    @Ctx() ctx: Context
+  ) {
+    const { token } = body
+    if (!token) return ctx.throw(badRequest('No token'))
```

Comment:
You'll always have token, validator won't accept body without it, we don't need this check